resources/scripts/generate-build-data.sh

```diff
@@ -163,6 +164,26 @@ function fetchAndPrepareTestsInfo() {
     echo "\"${key}\": $(cat "${file}")," >> "${BUILD_REPORT}"
 }
 
+function fetchAndPrepareTestCoberturaReport() {
```

Comment:
For same reason:
```suggestion
function fetchAndPrepareCodeCoverageReport() {
```

Reply:
I'm checking that Jenkins exposes a `/BUILD_URL/cobertura` endpoint. If that's true, that would mean that these changes should keep the naming convention of `cobertura`, as they seem coupled to the cobertura Jenkins plugin.

But what would happen with other code coverage tools, i.e. JaCoCo (which I think it's adding its own endpoint: https://github.com/jenkinsci/jacoco-plugin/blob/master/src/main/java/hudson/plugins/jacoco/JacocoProjectAction.java#L34?)?

Reply:
Unfortunately there is no one source of truth in terms for the `coverage` in Jenkins and every plugin provides their API, for instance, so far I identified the below two:

- `coverageReport` step, linked to the [cobertura](https://plugins.jenkins.io/cobertura/) plugin, uses the entrypoint `<build_URL>/cobertura/api/json?depth=2` for instance https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/master/1087/cobertura/api/json?depth=2
- `publishCoverage` step, linked to the [code coverage api](https://plugins.jenkins.io/code-coverage-api/) plugin, uses different entrypoints `<build_URL>/coverage/<COVERAGE_REPORT>​/result/api` for instance https://apm-ci.elastic.co/job/apm-agent-dotnet/job/apm-agent-dotnet-mbp/job/master/560/coverage/cobertura__01fb54b0_4583_4894_ad94_f62065adc97a_coverage_cobertura_xml/result/api/json?depth=1

We could normalise this somehow. Though, I went to the simpler path of creating one key in the document for each of the above, at the moment, there is only support for the `cobertura` plugin.

But if you think, we need to stop and think better this, let's do it :)

Reply:
With what you explained in https://github.com/elastic/apm-pipeline-library/pull/1214#discussion_r676359871, I'd see it convenient to rename to `test_coverage`, not to couple with the "cobertura" tool, using a generic word that could be the umbrella for any code coverage tool. And therefore, it's not needed to distinguish from code coverage.